source/common/tracing/http_tracer_impl.h

```diff
@@ -98,14 +100,29 @@ class HttpTracerUtility {
                             const Http::HeaderMap& request_headers);
 
   /**
-   * 1) Fill in span tags based on the response headers.
-   * 2) Finish active span.
+   * Fills in the active span that represents the downstream request with the proper headers.
+   * Then finishes the span.
    */
   static void finalizeSpan(Span& span, const Http::HeaderMap* request_headers,
                            const Http::HeaderMap* response_headers,
                            const Http::HeaderMap* response_trailers,
                            const StreamInfo::StreamInfo& stream_info, const Config& tracing_config);
 
+  /**
+   * Fills in the active span that represents each upstream request with the proper headers.
```

Comment:
"Adds tags to the active span that represent information obtained from the upstream request headers"?

Reply:
"Adds information obtained from the upstream request headers as tags to the active span."